opentelemetry-prometheus/src/lib.rs

```diff
@@ -240,6 +246,17 @@ impl ExporterBuilder {
         }
     }
 
+    /// Set the aggregation selector for the prometheus exporter
+    pub fn with_aggregator_selector(
+        self,
+        aggregator_selector: Box<dyn AggregatorSelector + Send + Sync>,
```

Comment:
Maybe we can take a type parameter `T` and wrap it using `Box` inside the function. May be more ergonomics?